src/databricks/labs/ucx/workspace_access/generic.py

```diff
@@ -324,11 +325,14 @@ def __iter__(self):
                 yield GenericPermissionsInfo(object_id=str(_object.object_id), request_type=request_type)
 
 
-def models_listing(ws: WorkspaceClient):
+def models_listing(ws: WorkspaceClient, num_threads: int):
     def inner() -> Iterator[ml.ModelDatabricks]:
-        for model in ws.model_registry.list_models():
-            model_with_id = ws.model_registry.get_model(model.name).registered_model_databricks
-            yield model_with_id
+        tasks = []
+        for m in ws.model_registry.list_models():
+            tasks.append(partial(ws.model_registry.get_model, name=m.name))
+        models, errors = Threads.gather("listing model ids", tasks, num_threads)
```

Comment:
and what do you do with errors? `/dev/null`? what if something fails?

throw ManyErrors.

https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/framework/parallel.py#L18-L22